processor/resourcedetectionprocessor/internal/aws/ec2/metadata.yaml

```diff
@@ -0,0 +1,39 @@
+type: ec2
```

Comment:
It may be good to include the parent in all of these subcomponents, e.g. `resourcedetectionprocessor/ec2`. This is what is done for subcomponents in the [hostmetricsreceiver for example](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/hostmetricsreceiver/internal/scraper/cpuscraper/metadata.yaml#LL1C17-L1C17). That said, this naming convention is [not required](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/cmd/mdatagen/metadata-schema.yaml#L1) -- so it can also stay like this.

Reply:
Ok, let's have it consistent, I added `resourcedetectionprocessor` prefix everywhere